### ISSUE TITLE
Model conversion between gpu and cpu

### ISSUE THREAD

**Issue by ZachzZ** [karpathy/char-rnn#67]
I try to do sampling with cpu using a model trained by gpu(cuz many online server don't have gpu). The original code has some problems to do this, and I fix them as follow:

First, convert the model to cpu model (you may do this before save)
```lua
checkpoint = torch.load(gpu_model)
checkpoint.protos.rnn:float()
checkpoint.protos.criterion:float()
torch.save(cpu_model, checkpoint)
```

Second, modify one line of sample.lua:
```lua
local h_init = torch.zeros(1, checkpoint.opt.rnn_size)
```
--->
```lua
local h_init = torch.zeros(1, checkpoint.opt.rnn_size):float()
```

May this help you.

**Comment by ZachzZ** [comment]
The conversion was included in the master branch 86a8eddbb8822bdcf4e42689dfab907c3bd59929 now.

**Comment by karpathy** [comment]
Yep, thanks @ZachzZ !
